buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -887,15 +889,7 @@ class ClusterFormationTasks {
             onlyIf { node.pidFile.exists() }
             // the pid file won't actually be read until execution time, since the read is wrapped within an inner closure of the GString
             ext.pid = "${ -> node.pidFile.getText('UTF-8').trim()}"
-            File jps
-            if (Os.isFamily(Os.FAMILY_WINDOWS)) {
-                jps = getJpsExecutableByName(project, "jps.exe")
-            } else {
-                jps = getJpsExecutableByName(project, "jps")
-            }
-            if (!jps.exists()) {
-                throw new GradleException("jps executable not found; ensure that you're running Gradle with the JDK rather than the JRE")
-            }
+            final File jps = Jvm.forHome(project.runtimeJavaHome).getExecutable('jps')
```

Comment:
So `project.runtimeJavaHome` may or may not be Java home used to run the build. Since `jps` in this case is a build dependency, it seems to make sense to use the Java runtime that Gradle is using via `Jvm.current()`. Realistically, we should be able to set `RUNTIME_JAVA_HOME` to a JRE, since it's just used for execution bits, not anything build related.

Reply:
>So `project.runtimeJavaHome` may or may not be Java home used to run the build.

Yup. The way I’ve always thought of it is:
 - we have the JVM running Gradle 
 - we have the JVM compiling code (compiler Java home)
 - we have the JVM running tests and all other executions that require a JVM (runtime Java home)

It’s not perfect (I think third-party audit runs in the Gradle JVM, although we want to move it out). I’d rather not break this mental model more though.

Reply:
Ah, giving this another think this makes sense. We should use the same version of `jps` that's used to launch any ES processes, even though it's actually the build that's executing `jps` here.

Reply:
Precisely.

Reply:
`ThirdPartyAuditTask` does have a `javaHome` property which we set to `runtimeJavaHome`.
`CheckForbiddenApis` runs in the same jvm ( does not work ) and has  a target compatibility specific 
to `runtimeJavaVersion`

Reply:
Thanks for the correction @atorok, I knew there was a `precommit` task that runs in the Gradle JVM that we ideally would want running in the runtime Java home JVM, I couldn't remember which though.

Reply:
We actually are slowly migrating to the opposite @jasontedor. There's not reason for precommit checks to run in the runtime JVM with `javac`, checkstyle and most other build-time verification runs using the Gradle JVM. These are "build" operations, not runtime ones so realistically they should run the the Gradle JVM. Also, forking them has a significant build time penalty as we are creating and throwing away lots of JVMs for each of these tasks.

This is a separate issue though and I think the usage of `runtimeJavaHome` is appropriate in this case. For others, we'll address them case-by-case but we are definitely misusing it in several places.